test/test_images/envvars/README.md

```diff
@@ -1,10 +1,12 @@
-# Helloworld test image
+# Envvars test image
 
 This directory contains the test image used in the envvars e2e test.
 
-The image contains a simple Go webserver, `envvars.go`, that will, by default, listen on port `8080` and expose a service at `/`.
+The image contains a simple Go webserver, `envvars.go`, that will, by default, listen on port `8080` and expose call to paths /envvars/must and /envvars/should `.
 
-When called, the server emits a message containing the values of the environment variables for knative service, configuraiton and revision.
+'must' and 'should' refer to the keywords "MUST" and "SHOULD" used in the Knative [run-time contract](/docs/runtime-contract.md) and provides environment variables that "MUST" and "SHOULD" be set by a Knative implementations
```

Comment:
I'd prefer more generic interfaces on our test containers (e.g. one container that provides `/envvars`, `/grow`, `/headers`, etc) rather than having a bunch of purpose-built containers, one for each test.

This would allow better re-use of the container code where we didn't need to write a new container for each new test.

Reply:
Makes sense, I have updated the test-image to be an environment test image that can be used to retrieve information about the environment in which the container runs. The server now exposes "/envvars" to fetch all the environment variables defined inside the container.